src/librustdoc/clean/inline.rs

```diff
@@ -290,10 +290,12 @@ pub fn build_ty(cx: &DocContext<'_>, did: DefId) -> Option<clean::Type> {
     }
 }
 
+// Passed `DefId` has to be a Trait item/path/reexport.
 pub fn build_impls(cx: &DocContext<'_>, did: DefId, attrs: Option<Attrs<'_>>) -> Vec<clean::Item> {
     let tcx = cx.tcx;
     let mut impls = Vec::new();
 
+    // for each implementation of a trait, build the clean::Item for that impl
     for &did in tcx.inherent_impls(did).iter() {
```

Comment:
Hmm, this doesn't look right. Inherent impls are things like `impl T { ... }`, not `impl Trait for T { ... }`. So the DefId would have to be a type, not a trait.

Normally I'd say to put the comments on `fn inherent_impls`, not here, but it looks like they're already there on both [the implementation](https://github.com/rust-lang/rust/blob/master/compiler/rustc_typeck/src/coherence/inherent_impls.rs#L29-L30) and [declaration](https://github.com/rust-lang/rust/blob/master/compiler/rustc_middle/src/query/mod.rs#L501-L509) ... @eddyb do you know how to add comments to a query so they'll show up in the generated docs? In particular [`TyCtxt::inherent_impls`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.TyCtxt.html#method.inherent_impls) is not very helpful.

Reply:
Opened https://github.com/rust-lang/rust/pull/76820 fixing this.

Reply:
> Inherent impls are things like `impl T { ... }`, not `impl Trait for T { ... }`.

Is it right because we have `for_` binding in `build_impl` function ?

Reply:
Now _I'm_ confused - does `inherent_impl` refer to both? 😕

Reply:
@GuillaumeGomez do you know what's going on here?

Reply:
You can have inherent impls on traits. ;)